src/Proxy/ContainerProxy.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Yiisoft\Yii\Debug\Proxy;
+
+use Psr\Container\ContainerInterface;
+use Psr\Container\ContainerExceptionInterface;
+use Yiisoft\Di\Container;
+use Yiisoft\Di\ContainerDelegateInterface;
+
+final class ContainerProxy extends ContainerInterfaceProxy implements ContainerDelegateInterface
+{
+    public function __construct(
+        ContainerInterface $container,
```

Comment:
Since concrete Yii DI container is used, it should be type-hinted as such:

```suggestion
        Container $container,
```

Reply:
Any PSR-11 container may be used with this proxy.

Reply:
How? `set()`, `setMultiple()`, `addProvider()`, `withRootContainer()` aren't part of `ContainerInterface` contract so likely there will be no such methods in another implementation of container.

Reply:
See checkNativeContainer() method